pypeit/core/skysub.py

```diff
@@ -470,146 +471,178 @@ def optimal_bkpts(bkpts_optimal, bsp_min, piximg, sampmask, samp_frac=0.80,
+        extraction using the FWHM determined from object finding (or provided by
+        the user) for the spatial profile. 
+    sn_gauss : int or float, default = 4.0
+        The signal to noise threshold above which optimal extraction
+        with non-parametric b-spline fits to the objects spatial
+        profile will be performed. For objects with median S/N <
+        sn_gauss, a Gaussian profile will simply be assumed because
+        there is not enough S/N to justify performing a more
+        complicated fit.
+    model_full_slit : bool, default = False
+        Set the maskwidth of the objects to be equal to the slit
+        width/2 such that the entire slit will be modeled by the
+        local skysubtraction. This mode is recommended for echelle
+        spectra with reasonably narrow slits.
+    model_noise : bool, default = True
+        If True, construct and iteratively update a model inverse variance image
+        using :func:`~pypeit.core.procimg.variance_model`.  Construction of the
+        model variance *requires* ``base_var``, and will use the provided values
+        or defaults for the remaining 
+        :func:`~pypeit.core.procimg.variance_model` parameters.  If False, a
+        variance model will not be created and instead the input sciivar will
+        always be taken to be the inverse variance. Note that in order for the
+        noise model to make any sense one needs to be subtracting the sky and
+        *not* the sky residuals. In other words, for near-IR reductions where
+        difference imaging has been performed and this algorithm is used to fit
+        out the sky residuals (but not the sky itself) one should definitely set
+        model_noise=False since otherwise the code will attempt to create a
+        noise model using sky residuals instead of the sky, which is incorrect
+        (does not have the right count levels).  In principle this could be
+        improved if the user could pass in a model of what the sky is for
+        near-IR difference imaging + residual subtraction
+    show_profile : bool, default=False
+        Show QA for the object profile fitting to the screen. Note
+        that this will show interactive matplotlib plots which will
+        block the execution of the code until the window is closed.
+    show_resids : bool, optional
+        Show the model fits and residuals.
+    use_2dmodel_mask : bool, optional
+        Use the mask made from profile fitting when extracting?
+    no_local_sky : bool, optional
+        If True, do not fit local sky model, only object profile and extract optimally
+        The objimage will be all zeros.
+    base_var : `numpy.ndarray`_, shape is (nspec, nspat), optional
+        The "base-level" variance in the data set by the detector properties and
+        the image processing steps.  See
+        :func:`~pypeit.core.procimg.base_variance`.
+    count_scale : :obj:`float`, `numpy.ndarray`_, optional
```

Comment:
I still am not following why counts_scale is required here. This should have already been multiplied into base_var. Then given that we model the "scaled" counts in this routine, I don't think this routine ever needs counts_scale? Or am I missing something?

Reply:
Resolved over zoom discussion.